mono/metadata/gc.c

```diff
@@ -70,6 +70,7 @@ gchar **mono_do_not_finalize_class_names ;
 #define mono_finalizer_unlock() mono_coop_mutex_unlock (&finalizer_mutex)
 static MonoCoopMutex finalizer_mutex;
 static MonoCoopMutex reference_queue_mutex;
+static gint32 reference_queue_mutex_inited = 0x0;
```

Comment:
No need to init.

Reply:
There is a need to init. The reference queue relies on the mutex.

Reply:
There is no need for this zeroing. It is how variables are already initialized. Like the finalizer_mutex, reference_queue_mutex right above.

Reply:
We should seriously consider that reference_queue_mutex is a pointer,  its nullness is a lack of init, and cas the ptr,  and  avoid  the separate boolean. The separate booleans tend to create a subtle race condition that people do not see. Merging the init-ness into the state tends to fix it.